### PR TITLE
Convert to App Router and Next.js 14.

### PR DESCRIPTION
**Pull request by leerob**
- Converts to TypeScript
- Use `async` / `await` for data fetching, when possible
- Use Metadata API
- Use built-in layouts support through `layout.tsx`
- Use built-in instant loading UI through `loading.tsx`
- Use co-located files inside `app` directory (e.g. styles)

### CODE REVIEW

**Comment by vercel-team** [comment]
#### 📝 Changed routes:



- [**/api/notes/\[id\]**](https://next-rsc-notes-n63uurmnr.vercel.sh/api/notes/[id])
- [**/api/notes**](https://next-rsc-notes-n63uurmnr.vercel.sh/api/notes)

4 deleted routes:
- **/index.server**
- **/note/[id].server**
- **/note/edit/[id].server**
- **/note/edit/index.server**

Commit 6845fba1ee045ef6074da9a32120b43c353ea041 (https://next-rsc-notes-n63uurmnr.vercel.sh).

File: app/layout.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+function Sidebar({ children }) {
+  return (
+    <>
+      <input type="checkbox" className="sidebar-toggle" id="sidebar-toggle" />
+      <section className="col sidebar">
+        <Link href={'/'} className="link--unstyled">
+          <section className="sidebar-header">
+            <img
```
Comment (Serj10GR) [inline app/layout.tsx:65]: just interesting to me why not next/image here?

**Comment by leerob** [comment]
Adding and editing seems to work. Still need to fix getting all notes and then searching.

**Comment by vercel-team** [comment]
#### 📝 Changed routes:




6 deleted routes:
- **/api/notes/[id]**
- **/api/notes**
- **/index.server**
- **/note/[id].server**
- **/note/edit/[id].server**
- **/note/edit/index.server**

Commit fe7229e70d4f9bcdf86cac75099212a9c75da160 (https://next-rsc-notes-jxurh9iv6.vercel.sh).

**Pull request merged**